Спринт 15

KateSilencio opened the pull request:
Версия 1

KateSilencio committed:
Спринт 15
Версия 1

IliaAnK commented on app/src/main/java/com/example/playlistmaker/ui/media/MediaActivity.kt:
```diff
@@ -32,9 +35,17 @@ class MediaActivity : AppCompatActivity() {
     private var playerState = STATE_DEFAULT
     private var mediaPlayer = MediaPlayer()
```
Обязательно:
`MediaPlayer` это своего рода источник данных. А с источниками данных должен работать репозиторий, а не `Activity`. Нужно выделить работу с `MediaPlayer` в `Repository` и не забыть про domain слой через который `Activity` будет обращаться к нему.

IliaAnK commented on app/src/main/java/com/example/playlistmaker/ui/search/SearchActivity.kt:
```diff
@@ -95,12 +97,9 @@ class SearchActivity : AppCompatActivity() {
         //Нажатие основной список поиска
         adapter.setOnClickListener { track ->
-            trackListHistory = history.mainLogicSaveTracks(track)
+            trackListHistory = historyRepository.saveTrack(track)
```
Обязательно:
`Activity` не должна быть напрямую связана с `data` слоем. Нужно добавить сущность `domain` слоя.

IliaAnK commented on app/src/main/java/com/example/playlistmaker/data/sharedprefs/SearchHistoryRepositoryImpl.kt:
```diff
@@ -0,0 +1,72 @@
+import com.google.gson.reflect.TypeToken
+import java.util.LinkedList
+
+class SearchHistoryRepositoryImpl(private val sharedPref: SharedPrefFunRepositoryImpl) :
```
Рекомендация:
1. Как правило, в качестве зависимости внедряется абстракция, а не имплементация, стоит заменить на `SearchHistoryLogicRepository`;
2. Не принято использовать репозиторий в репозитории.
Как вариант, `SharedPrefFunRepositoryImpl` можно переименовать в `SharedPrefFunLocalDataSource`.

KateSilencio committed:
Спринт 15
Версия 2
Исправлено:
1. Работа с MediaPlayer вынесена в слой data
2. Добавлен Use case SearchHistoryUseCase для работы с searchHistory, который используется в SearchActivity
3. В SearchHistoryRepositoryImpl используется интерфейс SharedPrefFunRepository вместо имплементации репозитория

IliaAnK commented on app/src/main/java/com/example/playlistmaker/ui/media/MediaActivity.kt:
```diff
@@ -30,11 +34,22 @@ class MediaActivity : AppCompatActivity() {
-    //private lateinit var track: TracksFields
-    private var play: ImageView?= null
+    private val mediaPlayer: MediaPlayerRepository by lazy {
+        Creator.provideMediaPlayerRepository()
```
Обязательно:
`Activity` не должна быть напрямую связана с `data` слоем. Нужно добавить сущность `domain` слоя.

KateSilencio committed:
Спринт 15
Версия 3
Исправлено:
1. Добавлена сущность MediaPlayerUseCase в слой domain для работы в MediaActivity

IliaAnK commented on app/src/main/java/com/example/playlistmaker/data/media/MediaPlayerRepositoryImpl.kt:
```diff
@@ -0,0 +1,41 @@
+
+class MediaPlayerRepositoryImpl: MediaPlayerRepository {
+
+    private val mediaPlayer = MediaPlayer()
```
Рекомендация:
Это зависимость, которую можно создавать в `Creator` и передавать сюда через конструктор класса.

IliaAnK commented on app/src/main/java/com/example/playlistmaker/domain/api/media/usecase/MediaPlayerUseCaseImpl.kt:
```diff
@@ -0,0 +1,34 @@
+
+import com.example.playlistmaker.domain.api.media.MediaPlayerRepository
+
+class MediaPlayerUseCaseImpl(private val mediaPlayerRepository: MediaPlayerRepository):MediaPlayerUseCase {
```
Рекомендация:
`UseCase` - это "плоская" сущность `domain` слоя, в которой имеет только один метод внутри себя.
Всё остальное отличное от этого называется `Interactor`
